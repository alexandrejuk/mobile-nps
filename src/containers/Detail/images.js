// equipments
import A630 from '../../assets/images/equipments/a6-30.png'
import A860 from '../../assets/images/equipments/a8-60.png'
import ATP60E from '../../assets/images/equipments/atp60e.png'
import FTP60E from '../../assets/images/equipments/ftp60e.png'
import FTS60E from '../../assets/images/equipments/fts60e.png'
import Lioness from '../../assets/images/equipments/lioness.png'
import MTS60E from '../../assets/images/equipments/mts60e.png'
import PrimaPremix from '../../assets/images/equipments/prima-premix.png'
import M860 from '../../assets/images/equipments/m8-60.png'

// revenues image
import AO140mlLP from '../../assets/images/revenues/AO140mlLP.png'
import AO90mlLP from '../../assets/images/revenues/AO90mlLP.png'
import AP140mlLP from '../../assets/images/revenues/AP140mlLP.png'
import AP90mlLP from '../../assets/images/revenues/AP90mlLP.png'
import X140ml from '../../assets/images/revenues/X140ml.png'
import X90ml from '../../assets/images/revenues/X90ml.png'
import AM140ml from '../../assets/images/revenues/AM140ml.png'
import AM90ml from '../../assets/images/revenues/AM90ml.png'
import AO140ml from '../../assets/images/revenues/AO140ml.png'
import AO90ml from '../../assets/images/revenues/AO90ml.png'
import AP140ml from '../../assets/images/revenues/AP140ml.png'
import AP90ml from '../../assets/images/revenues/AP90ml.png'
import C018 from '../../assets/images/revenues/C018.png'
import BM860 from '../../assets/images/revenues/BM860.png'
import EM860 from '../../assets/images/revenues/EM860.png'
import AI140ml from '../../assets/images/revenues/AI140ml.png'
import AI90ml from '../../assets/images/revenues/AI90ml.png'
import C001 from '../../assets/images/revenues/C001.png'
import HMTS60E from '../../assets/images/revenues/HMTS60E.png'
import HMenu2MTS60E from '../../assets/images/revenues/HMenu2MTS60E.png'
import FMTS60E from '../../assets/images/revenues/FMTS60E.png'
import IMTS60E from '../../assets/images/revenues/IMTS60E.png'
import GMTS60E from '../../assets/images/revenues/GMTS60E.png'
import JMTS60E from '../../assets/images/revenues/JMTS60E.png'
import C009GPA from '../../assets/images/revenues/C009GPA.png'
import C009GPAMenu2 from '../../assets/images/revenues/C009GPAMenu2.png'
import C010Minuto from '../../assets/images/revenues/C010Minuto.png'
import KMTS60E from '../../assets/images/revenues/KMTS60E.png'

const revenuesImage = {
  AO140mlLP,
  AO90mlLP,
  AP140mlLP,
  AP90mlLP,
  X140ml,
  X90ml,
  AM140ml,
  AM90ml,
  AO140ml,
  AO90ml,
  AP140ml,
  AP90ml,
  C018,
  BM860,
  EM860,
  AI140ml,
  AI90ml,
  C001,
  HMTS60E,
  HMenu2MTS60E,
  FMTS60E,
  IMTS60E,
  GMTS60E,
  JMTS60E,
  C009GPA,
  C009GPAMenu2,
  C010Minuto,
  KMTS60E,
}

const imagesProduct = {
  A630,
  A860,
  ATP60E,
  FTP60E,
  FTS60E,
  Lioness,
  MTS60E,
  PrimaPremix,
  M860,
}

export {
  revenuesImage,
  imagesProduct,
}
