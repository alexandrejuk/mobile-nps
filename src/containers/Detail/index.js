import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { map, prop } from 'ramda'
import Lightbox from 'react-image-lightbox'
import 'react-image-lightbox/style.css'

import {
  Collapse,
  Header,
  Title,
} from '../../components'

import styles from './style.module.css'

import {
  revenuesImage,
  imagesProduct,
} from './images'

const issue = ({
  id,
  name,
  problems,
  resolves,
}) => (
  <Collapse key={id} title={name}>
    <div className={styles.issueSection}>
      <Title
        size="small"
        text="Problema"
        type="bold"
      />
    {
      problems && map((problem, index) =>(
        <Title
          key={`${index}${problem}`}
          size="small"
          text={problem}
        />
      ), problems)
    }
    </div>
    <div className={styles.issueSection}>
      <Title
        size="small"
        text="Solução"
        type="bold"
      />
    {
      resolves && map((resolve, index) => (
        <Title
          key={`${index}${resolve}`}
          size="small"
          text={resolve}
        />
      ), resolves)
    }
     </div>
  </Collapse>
)

const Detail = ({
  product = {},
  goBack,
}) => {
  const [isOpen, setIsOpen] = useState(false)
  const [photoIndex, setPhotoIndex] = useState(0)
  const revenues = prop('revenues', product)
  const images = map(({ src }) => (revenuesImage[src]), revenues)

  const handleRevenue = (index) => {
    setPhotoIndex(index)
    setIsOpen(true)
  }

  const revenue = ({
    src,
    title,
  }, index) => (
    <div
      className={styles.revenueItem}
      key={index}
      onClick={() => handleRevenue(index)}
    >
      <Title
        size="small"
        text={`Cod: ${title}`}
        type="bold"
      />
      <div className={styles.revenuesAvatar}>
        <img src={revenuesImage[src]} alt="revenue gallery" />
      </div>
    </div>
  )
  return (
    <div className={styles.container}>
      <Header
        action={goBack}
      />
      <div className={styles.product}>
        <Title
          size="medium"
          text={prop('name', product)}
          textAlign="right"
        />
        <div className={styles.productImage}>
          <img src={imagesProduct[prop('image', product)]} alt="product" />
        </div>
        <div className={styles.productContent}>
          <Title
            size="medium"
            text="Descrição"
          />
          <div className={styles.productInfo}>
            <div className={styles.descriptionText}>
              <Title
                color="primary"
                size="small"
                text={prop('description', product)}
              />
            </div>
            <div className={styles.productInfoSection}>
              <Title
                color="primary"
                size="small"
                text="Dimensões"
              />
              <Title
                color="primary"
                size="small"
                type="bold"
                text={`A: ${prop('height', product)} x L: ${prop('width', product)} x P: ${prop('deep', product)}`}
              />
            </div>
            <div className={styles.equipmentTechnical}>
              <div className={styles.productInfoSection}>
                <Title
                  color="primary"
                  size="small"
                  text="Potência/Tensão"
                />
                <Title
                  color="primary"
                  size="small"
                  type="bold"
                  text={`${prop('potency', product)} / ${prop('tension', product)}`}
                />
              </div>
              <div className={styles.productInfoSection}>
                <Title
                  color="primary"
                  size="small"
                  text="Peso"
                />
                <Title
                  color="primary"
                  size="small"
                  type="bold"
                  text={`${prop('weight', product)}`}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className={styles.productRevenue}>
        <Title
          size="medium"
          text="Receitas"
        />
      </div>
      <div className={styles.productRevenueDocumet}>
        <div className={styles.revenueGallery}>
          {revenues.map(revenue)}
        </div>
        {
          isOpen && (
            <Lightbox
              mainSrc={images[photoIndex]}
              nextSrc={images[(photoIndex + 1) % images.length]}
              prevSrc={images[(photoIndex + images.length - 1) % images.length]}
              onCloseRequest={() => setIsOpen(false)}
              onMovePrevRequest={() =>
                setPhotoIndex((photoIndex + images.length - 1) % images.length)
              }
              onMoveNextRequest={() =>
                setPhotoIndex((photoIndex + 1) % images.length)
              }
            />
          )
        }
      </div>
      <div className={styles.productIssue}>
      {product.issues && (
          <div className={styles.productIssueTitle}>
            <Title
              size="medium"
              text="Erros"
            />
          </div>
        )}
        { product.issues && (
           map(issue, prop('issues', product))
        )}
      </div>
    </div>
  )
}

Detail.propTypes = {
  goBack: PropTypes.func.isRequired,
  product: PropTypes.shape({}).isRequired,
}

export default Detail
