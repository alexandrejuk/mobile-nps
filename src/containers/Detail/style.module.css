@import '../../assets/css/colors.css';

.container {
  background-color: var(--color-concrete);
}

.product {
  background-color: var(--color-white);
  border-radius: 3px;
  box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.15);
  padding: 20px;
}

.productImage {
  height: 229px;
  margin: auto;
  width: 215px;
}

.productImage img {
  height: 100%;
}

.productContent {
  padding-top: 30px;
}

.productInfo {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

.productInfoSection {
  flex: 1;
}

.productRevenue {
  padding: 10px 20px;
}

.productIssueTitle {
  padding: 20px;
}

.issueSection {
  padding: 10px 0;
}

.productRevenueDocumet {
  box-sizing: border-box;
  overflow: hidden;
  padding: 10px 20px;
  width: 100%;
}

.productRevenueDocumet img {
  width: 100%;
}

.revenuesAvatar {
  border-radius: 3px;
  height: 82px;
  overflow: hidden;
  width: 82px;
}

.revenuesAvatar img {
  height: 100%;
}

.revenueGallery {
  align-items: center;
  display: flex;;
  flex-wrap: wrap;
  justify-content: space-between;
}

.revenueItem {
  flex: 1;
  margin-bottom: 20px;
}
