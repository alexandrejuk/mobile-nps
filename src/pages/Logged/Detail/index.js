import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router'
import DetailContainer from '../../../containers/Detail'
import EquipmentsMock from '../../../utils/api/mocks'

import GAInitialize from '../../../utils/ga'

const Detail = ({
  history,
  match
}) => {
  const goBack = () => history.goBack()
  const { id } = match.params
  const product = EquipmentsMock.find(product => product.id === Number(id))
  GAInitialize(`/logged/detail/${id}`)
  return (
    <DetailContainer
      goBack={goBack}
      product={product || {}}
    />
  )
}

Detail.propTypes = {
  history: PropTypes.shape({
    goBack: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
      ]).isRequired,
    }).isRequired,
  }).isRequired
}

export default withRouter(Detail)
