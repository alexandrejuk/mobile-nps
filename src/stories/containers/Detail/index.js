import React from 'react'
import { action } from '@storybook/addon-actions'
import Detail from '../../../containers/Detail'

export default {
  title: 'Containers/Detail',
}

const product =   {
  description: 'Maquina de café premium, criada com o mais alto padrão de qualidade, lider de venda em mais de 2.000 mil países.',
  deep: 2400,
  id: 3,
  image: 'https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9',
  issues: [
    {
      id: 1,
      name: 'off 0',
      problem: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum?',
      resolve: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum.',
    },
    {
      id: 2,
      name: 'off 1',
      problem: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum?',
      resolve: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum.',
    },
  ],
  name: 'A6/30',
  height: 100,
  revenues: [
    {
      src: 'https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9',
      title: 'AL',
      description: 'Receita blabla',
    },
    {
      src: 'https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9',
      title: 'P',
      description: 'Receita blabla',
    },
  ],
  voltage: '220v~/110v~',
  width: 330,
  weight: 1.2,
}

export const Default = () => (
  <Detail
    goBack={action('go back!')}
    product={product}
  />
)
