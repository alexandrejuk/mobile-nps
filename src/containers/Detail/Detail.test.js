import React from 'react'
import { render } from '@testing-library/react'

import Detail from './'

const productMock = {
  description: 'Maquina de café premium, criada com o mais alto padrão de qualidade, lider de venda em mais de 2.000 mil países.',
  deep: 2400,
  id: 3,
  image: 'M860',
  issues: [
    {
      id: 1,
      name: 'off 0',
      problem: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum?',
      resolve: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum.',
    },
  ],
  name: 'A6/30',
  height: 100,
  revenues: [
    {
      id: 1,
      src: 'https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9',
      title: 'AL',
      description: 'Receita blabla',
    }
  ],
  potency: '220v~',
  tension: '1600w',
  width: 330,
  weight: 1.2,
}

describe('Detail container', () => {
  let DetailContainer
  it('should render correctly', () => {
    const callback = jest.fn()
    DetailContainer = render(
      <Detail
        goBack={callback}
        product={productMock}
        issueSelected={{}}
        showModal={false}
        onOpenModal={callback}
        onCloseModal={callback}
      />
    )

    const SampleEquimentCode = DetailContainer.getByText(productMock.name)
    const SampleImgSrc = DetailContainer.getByAltText('product')
    const SampleDescription = DetailContainer.getByText('Description of Problem')
    const SampleEquimentPotency = DetailContainer.getByText(`${productMock.potency} | ${productMock.tension}`)

    expect(SampleDescription).toBeInTheDocument()
    expect(SampleImgSrc.getAttribute('src')).toEqual('m8-60.png')
    expect(SampleEquimentCode).toBeInTheDocument()
    expect(SampleEquimentPotency).toBeInTheDocument()
    expect(DetailContainer.container).toMatchSnapshot()
  })
})
