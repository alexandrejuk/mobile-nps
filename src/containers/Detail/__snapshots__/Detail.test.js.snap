// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail container should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="icon"
        role="button"
      >
        <img
          alt="action go back"
          src="arrow-left.svg"
        />
      </div>
      <div
        class="logo"
      >
        <img
          alt="company logo"
          src="logo.svg"
        />
      </div>
    </div>
    <div
      class="product"
    >
      <h1
        class="primary medium right title normal"
      >
        A6/30
      </h1>
      <div
        class="productImage"
      >
        <img
          alt="product"
          src="m8-60.png"
        />
      </div>
      <div
        class="productContent"
      >
        <h1
          class="primary medium left title normal"
        >
          Descrição
        </h1>
        <div
          class="productInfo"
        >
          <div
            class="productInfoSection"
          >
            <h1
              class="primary small left title normal"
            >
              Tensão
            </h1>
            <h1
              class="primary small left title normal"
            >
              220v~/110v~
            </h1>
          </div>
          <div
            class="productInfoSection"
          >
            <h1
              class="primary small left title normal"
            >
              Dimensões
            </h1>
            <h1
              class="primary small left title normal"
            >
              AL: 100 x L: 330 x P: 2400 - Peso: 1.2
            </h1>
          </div>
        </div>
      </div>
    </div>
    <div
      class="productRevenue"
    >
      <h1
        class="primary medium left title normal"
      >
        Receitas
      </h1>
    </div>
    <div
      class="productRevenueDocumet"
    >
      <div
        class="revenueGallery"
      >
        <div
          class="revenueItem"
        >
          <h1
            class="primary small left title bold"
          >
            Cod: AL
          </h1>
          <div
            class="revenuesAvatar"
          >
            <img
              alt="revenue gallery"
              src="https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="productIssue"
    >
      <div
        class="productIssueTitle"
      >
        <h1
          class="primary medium left title normal"
        >
          Erros
        </h1>
      </div>
      <div
        class="collapse"
        style="height: 65px;"
      >
        <div
          class="collapseHeader"
          role="button"
          tabindex="0"
        >
          <h1
            class="primary medium left title normal"
          >
            off 0
          </h1>
          <div
            class="icon"
          >
            <img
              alt="arrow"
              src="arrow-up.svg"
            />
          </div>
        </div>
        <div
          class="collapseBody"
        >
          <div
            class="issueSection"
          >
            <h1
              class="primary small left title bold"
            >
              Problema
            </h1>
            <h1
              class="primary small left title normal"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum?
            </h1>
          </div>
          <div
            class="issueSection"
          >
            <h1
              class="primary small left title bold"
            >
              Solução
            </h1>
            <h1
              class="primary small left title normal"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Iusto sequi totam corporis cum.
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
